### PR TITLE
[ROCm] enumerate ROCm GPU devices in grappler.

### PR DESCRIPTION
**Pull request by whchung**
Co-authored with @jeffdaily .

Enable enumerating ROCm GPU devices constructing grappler clusters.

### CODE REVIEW

**Comment by rthadur** [comment]
@jaingaurav gentle ping for review

File: tensorflow/core/grappler/devices.cc [hunk 1]
```diff
@@ -19,18 +19,25 @@ limitations under the License.
...
+#endif  // GOOGLE_CUDA || TENSORFLOW_USE_ROCM
 
 namespace tensorflow {
 namespace grappler {
 
 int GetNumAvailableGPUs(
-    const std::pair<int, int>& min_cuda_compute_capability) {
+    const GpuVersion& min_gpu_version) {
   int num_eligible_gpus = 0;
+#if GOOGLE_CUDA || TENSORFLOW_USE_ROCM
```
Comment (jaingaurav) [inline tensorflow/core/grappler/devices.cc]: Using ifdef's in the code doesn't seem right here. Can we just have 3 separate files? device_default.cc, device_cuda.cc & device_rocm.cc and fix the build rules to include the correct one?

File: tensorflow/core/grappler/devices.h [hunk 2]
```diff
@@ -19,18 +19,33 @@ limitations under the License.
...
+// Get the number of available GPUs.
+// On CUDA platform, look for GPUs whose number of multiprocessors is no less
+// than 8 and whose CUDA compute capability is no less than min_gpu_version,
+// represented as a pair of integers.
+// On ROCm platform, look for GPUs whose ISA version number is no less than
+// min_gpu_version, represented as a single integer.
 int GetNumAvailableGPUs(
-    const std::pair<int, int>& min_cuda_compute_capability = {0, 0});
+#if GOOGLE_CUDA
+    const GpuVersion& min_gpu_version = std::pair<int, int>(0, 0)
```
Comment (jaingaurav) [inline tensorflow/core/grappler/devices.h]: Same comment goes here. I don't think we should have preprocessor macros that change the function signature. This should be broken out into different files.

**Review by jaingaurav — CHANGES_REQUESTED** [review]

**Comment by gbaned** [comment]
@whchung  Could you please check reviewer comments and keep us posted. Thanks!

**Comment by whchung** [comment]
Closing this PR as from empirical testing it doesn't bring merits.

**Pull request closed without merging**